### PR TITLE
Add python tests to Makefile & workflows

### PR DESCRIPTION
**Pull request by fmoletta**
Add targets `run-python-tests`, and `full-test` which runs both python and rust tests
Add targets `run-python-tests-macos`, and `full-test-macos` which also runs the command that adds the missing flags 
Add targets `run-python-tests-default-version`, and `full-test-default-version` which don't set the peen version (meant to be used by workflow)
 
Swap `make test` for `make full-test` in workflow

### CODE REVIEW

**Review by Juan-M-V — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -29,3 +29,39 @@ clippy:
...
 	rm -f $(TEST_DIR)/*.json
+	rm -rf cairo-rs-py-env
+
+run-python-test-macos: $(COMPILED_TESTS)
+	python3 -m venv cairo-rs-py-env
+	. cairo-rs-py-env/bin/activate && \
+	pyenv install pypy3.7-7.3.9 && \
+	pyenv local pypy3.7-7.3.9 && \
+	CFLAGS=-I/opt/homebrew/opt/gmp/include LDFLAGS=-L/opt/homebrew/opt/gmp/lib pip install fastecdsa && \
+	pip install cairo_lang==0.9.1 && \
```
Comment (Oppen) [inline Makefile:40]: All of this belongs to a `deps` target. Also, the `&& \` is not necessary, any `make` target fails if any of its commands fails.

Comment (Oppen) [reply Makefile:40]: Also, don't set `local`, it's more robust to pass the `PYENV_VERSION` environmental variable instead.

Comment (Oppen) [reply Makefile:40]: `venv` activation does belong here tho. Deactivation requires a little bit more care for failing tests.

File: Makefile [hunk 2]
```diff
@@ -29,3 +29,39 @@ clippy:
...
+full-test-macos: test run-python-test-macos clean
+
+run-python-test: $(COMPILED_TESTS)
+	python3 -m venv cairo-rs-py-env
+	. cairo-rs-py-env/bin/activate && \
+	pyenv install pypy3.7-7.3.9 && \
+	pyenv local pypy3.7-7.3.9 && \
+	pip install cairo_lang==0.9.1 && \
+	maturin develop && \
+	python3 hints_tests.py && \
```
Comment (Oppen) [inline Makefile:54]: Ditto.

File: Makefile [hunk 3]
```diff
@@ -29,3 +29,39 @@ clippy:
...
+	deactivate
+
+full-test: test run-python-test clean
+
+run-python-test-default-version: $(COMPILED_TESTS)
+	python3 -m venv cairo-rs-py-env
+	. cairo-rs-py-env/bin/activate && \
+	pip install cairo_lang==0.9.1 && \
+	maturin develop && \
+	python3 hints_tests.py && \
```
Comment (Oppen) [inline Makefile:64]: Ditto.

**Review by Oppen — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (fmoletta) [reply Makefile]: Moved env setup to deps
The reason for using `&& \ `is that if not used then the virtual environment is not activated when `maturin develop` is called, causing failure.

**Review by Oppen — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -8,21 +8,22 @@ $(TEST_DIR)/%.json: $(TEST_DIR)/%.cairo
...
-	pip install cairo_lang==0.9.1
 
 deps-macos:
+	CFLAGS=-I/opt/homebrew/opt/gmp/include LDFLAGS=-L/opt/homebrew/opt/gmp/lib pip install fastecdsa
+	pip install ecdsa fastecdsa sympy cairo-lang==0.9.1 maturin
 	python3 -m venv cairo-rs-py-env
 	pyenv install pypy3.7-7.3.9
 	export PYENV_VERSION=pypy3.7-7.3.9
-	CFLAGS=-I/opt/homebrew/opt/gmp/include LDFLAGS=-L/opt/homebrew/opt/gmp/lib pip install fastecdsa
-	pip install cairo_lang==0.9.1
```
Comment (Oppen) [inline Makefile]: Why was this moved back into the test targets?

Comment (fmoletta) [reply Makefile]: We need to install `cairo_lang ` inside the virtual environment, otherwise when running cairo_rs_py, imports from starkware files are not recognized

Comment (fmoletta) [reply Makefile]: We need to install `cairo_lang` inside the virtual environment, otherwise cairo_rs_py is not able to recognize imports from starkware (aka importing stuff from starkware files inside hints)

Comment (Oppen) [reply Makefile]: But doesn't that belong to the env creation?

**Review by Oppen — APPROVED** [review]
